packages/svelte/src/lib/useForm.ts

```diff
@@ -0,0 +1,278 @@
+import type {
+  ActiveVisit,
+  Errors,
+  Method,
+  Page,
+  PendingVisit,
+  Progress,
+  RequestPayload,
+  VisitOptions,
+} from '@inertiajs/core'
+import { router } from '@inertiajs/core'
+import type { AxiosProgressEvent } from 'axios'
+import cloneDeep from 'lodash/cloneDeep'
+import isEqual from 'lodash/isEqual'
+import { writable, type Writable } from 'svelte/store'
+
+interface InertiaFormProps<TForm extends Record<string, unknown>> {
+  isDirty: boolean
+  errors: Partial<Record<keyof TForm, string>>
+  hasErrors: boolean
+  progress: Progress | null
+  wasSuccessful: boolean
+  recentlySuccessful: boolean
+  processing: boolean
+  setStore(data: InertiaFormProps<TForm>): void
+  setStore(key: keyof InertiaFormProps<TForm>, value?: any): void
```

Comment:
I believe `setStore` should not include the attributes or methods of the form. Should be typed to the data.
In my branch I did
```
  setStore(data: TForm): void
  setStore(key: keyof TForm, value?: FormDataConvertible): void
  defaults(): this
  defaults(fields: Partial<TForm>): this
  defaults(field?: keyof TForm, value?: FormDataConvertible): this
```